compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -305,6 +306,45 @@ void LinalgFusePass::runOnOperation() {
   }
 }
 
+/// Return true if 'loop' is a tiled loop that distributes work among threads.
+/// We currently check whether the loop control depends on any of the interface
+/// workgroup operations. This may lead to some false positives that may prevent
+/// peeling of loops that we could peel. We shouldn't need this check when we
+/// move to the Transform dialect.
```

Comment:
The main problem with the selection of loops to peel is that we may tile loops for vectorization that are later removed if they have a single iteration. (IMO, we should be doing this kind of loop optimizations at the end and not in the middle of the strategy but that's a discussion for another day 😄). The current implementation seems to be enough and fall on the conservative side but I'm open to other suggestions.

Reply:
Yeah, that is a issue with throwing everything into the strategy.

But you should be able to use this (does a similar traversal) https://github.com/google/iree/blob/e2876535b5208f012b97b92c8a058ae9803d48ac/compiler/src/iree/compiler/Codegen/Utils/Utils.h#L160